ksql-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlAvroDeserializerTest.java

```diff
@@ -53,236 +62,733 @@
-    shouldDeserializeTypeCorrectly(avroSchema, avroValue, ksqlSchema, avroValue);
+  @Test
+  public void shouldDeserializedAvroBoolean() {
+    // Given:
+    givenDeserializerForSchema(Schema.OPTIONAL_BOOLEAN_SCHEMA);
+
+    final byte[] bytes = givenAvroSerialized(true, BOOLEAN_AVRO_SCHEMA);
+
+    // When:
+    final Object result = deserializer.deserialize(SOME_TOPIC, bytes);
+
+    // Then:
+    assertThat(result, is(true));
   }
 
-  private void shouldDeserializeTypeCorrectly(final org.apache.avro.Schema avroSchema,
-                                              final Object avroValue,
-                                              final Schema ksqlSchema,
-                                              final Object ksqlValue) {
+  @Test
+  public void shouldThrowIfCanNotCoerceToBoolean() {
+    // Given:
+    givenDeserializerForSchema(Schema.OPTIONAL_BOOLEAN_SCHEMA);
 
+    final byte[] bytes = givenAvroSerialized(10, LONG_AVRO_SCHEMA);
 
-    final org.apache.avro.Schema avroRecordSchema = org.apache.avro.SchemaBuilder.record("test_row")
-        .fields()
-        .name("field0")
-        .type(avroSchema)
-        .noDefault()
-        .endRecord();
-    final Schema ksqlRecordSchema = SchemaBuilder.struct().field("field0", ksqlSchema).build();
+    // Then:
+    expectedException.expect(SerializationException.class);
+    expectedException.expectCause(hasMessage(containsString(
+        "Cannot deserialize type int32 as type boolean")));
 
-    final GenericRecord avroRecord = new GenericData.Record(avroRecordSchema);
-    avroRecord.put("field0", avroValue);
+    // When:
+    deserializer.deserialize(SOME_TOPIC, bytes);
+  }
 
-    final Struct row = serializeDeserializeAvroRecord(
-        ksqlRecordSchema, "test-topic", schemaRegistryClient, avroRecord);
+  @Test
+  public void shouldDeserializedAvroInt() {
+    // Given:
+    givenDeserializerForSchema(Schema.OPTIONAL_INT32_SCHEMA);
 
-    assertThat(row.schema(), is(ksqlRecordSchema));
-    assertThat(row.get("field0"), equalTo(ksqlValue));
+    final byte[] bytes = givenAvroSerialized(42, INT_AVRO_SCHEMA);
+
+    // When:
+    final Object result = deserializer.deserialize(SOME_TOPIC, bytes);
+
+    // Then:
+    assertThat(result, is(42));
+  }
+
+  @Test
+  public void shouldThrowIfCanNotCoerceToInt() {
+    // Given:
+    givenDeserializerForSchema(Schema.OPTIONAL_INT32_SCHEMA);
+
+    final byte[] bytes = givenAvroSerialized(true, BOOLEAN_AVRO_SCHEMA);
+
+    // Then:
+    expectedException.expect(SerializationException.class);
+    expectedException.expectCause(hasMessage(containsString(
+        "Cannot deserialize type boolean as type int32")));
+
+    // When:
+    deserializer.deserialize(SOME_TOPIC, bytes);
+  }
+
+  @Test
+  public void shouldDeserializedAvroBigInt() {
+    // Given:
+    givenDeserializerForSchema(Schema.OPTIONAL_INT64_SCHEMA);
+
+    final Map<org.apache.avro.Schema, Object> validCoercions = ImmutableMap.of(
```

Comment:
Should include int8, int16

Reply:
Avro doesn't have such types.  This test is _deserializing_ an avro type to Connect INT64. The only types we support deserializing into a BIGINT are Avro `int` and `long` if I'm not mistaken.

Reply:
> We don't want to ever deserialize to an int8/16...

Connect can generate such types. E.g. I can use the connect avro converter to serialize a Schema.INT8_SCHEMA. Our tests should verify that we deserialize these correctly.

Reply:
That doesn't make sense to me.  This is the deserializer test.  It tests we can deserialize avro types to the connect ones.   A test that checks we can serialize an INT8 into an avro int is testing the serializer, not the deserializer.

Reply:
Let's say I'm using Connect to write an int8 to a Kafka topic from some external source. Connect will write that data out as an avro int, and add some metadata to the avro schema describing the original type (int8). We should have a test that verifies that we can deserialize what Connect has written out into a KSQL field (of type INTEGER or BIGINT (or SHORT/INT16 or BYTE/INT8 in the future).

Reply:
Ah - gotcha!  It was the metadata that I was missing.  Consider it done!

Reply:
Done in second commit to #2941